### ISSUE TITLE
node fails to start after service stop - seastar - Exiting on unhandled exception: exceptions::read_timeout_exception (similar symptoms: #2943 or #3015) 

### ISSUE THREAD

**Issue by amuzychka** [scylladb/scylla#3138]
*Installation details*
Scylla version: 1.7.5-0.20170918.760af56
Cluster size: 5x1TB
AWS AMI: ami-9bc9bd8d
Platform: i3.2xlarge

This issue can be related to #2943 or #3015 but for Scylla version 1.7

In case of high load, random nodes in cluster failed and can't start again.

**journalctl -xe log**

```
jan 20 09:28:39 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 0] seastar - Exiting on unhandled exception: exceptions::read_timeout_exception (Operation timed out for system.local - received only 0 responses from 1 CL=ONE.)
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 0] compaction_manager - Asked to stop
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 5] compaction_manager - Asked to stop
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 3] compaction_manager - Asked to stop
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 5] compaction_manager - Stopped
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 6] compaction_manager - Asked to stop
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 3] compaction_manager - Stopped
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 6] compaction_manager - Stopped
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 4] compaction_manager - Asked to stop
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 1] compaction_manager - Asked to stop
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 4] compaction_manager - Stopped
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 1] compaction_manager - Stopped
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 2] compaction_manager - Asked to stop
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 2] compaction_manager - Stopped
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws scylla[5992]:  [shard 0] compaction_manager - Stopped
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws systemd[1]: scylla-server.service: main process exited, code=exited, status=1/FAILURE
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws systemd[1]: Failed to start Scylla Server.
-- Subject: Unit scylla-server.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit scylla-server.service has failed.
-- 
-- The result is failed.
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws systemd[1]: Dependency failed for Scylla JMX.
-- Subject: Unit scylla-jmx.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit scylla-jmx.service has failed.
-- 
-- The result is dependency.
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws systemd[1]: Job scylla-jmx.service/start failed with result 'dependency'.
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws systemd[1]: Dependency failed for Run Scylla Housekeeping daily mode.
-- Subject: Unit scylla-housekeeping-daily.timer has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit scylla-housekeeping-daily.timer has failed.
-- 
-- The result is dependency.
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws systemd[1]: Job scylla-housekeeping-daily.timer/start failed with result 'dependency'.
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws systemd[1]: Dependency failed for Run Scylla Housekeeping restart mode.
-- Subject: Unit scylla-housekeeping-restart.timer has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit scylla-housekeeping-restart.timer has failed.
-- 
-- The result is dependency.
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws systemd[1]: Job scylla-housekeeping-restart.timer/start failed with result 'dependency'.
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws systemd[1]: Unit scylla-server.service entered failed state.
jan 20 09:28:41 ip-10-124-43-5.us-east-1.aws systemd[1]: scylla-server.service failed.
```

node config example: [scylla.yaml.zip](https://github.com/scylladb/scylla/files/1648703/scylla.yaml.zip)

**Comment by slivne** [comment]
@amuzychka - it seems you have two issues

nodes are failing - can you open an issue with the ifnormation about your nodes failing and provide all the information you have 
if you can please follow: http://docs.scylladb.com/operating-scylla/troubleshooting/report_scylla_problem/

With regards to nodes failing to boot it will help if you include all the info from the start of the boot and not just the end (indeed it seems similar to #3015)

**Comment by amuzychka** [comment]
@slivne I already clean up that cluster so I can't provide logs, but looks like I found what was happening (at least in my case).
The reason why Scylla crashed was in our data pipeline. Scylla failed when 12 Hadoop clusters started **simultaneous** writes to 12 difference keyspaces, db failed in few first seconds, a simple delay in 10 seconds for each Hadoop cluster fixed this.

And 2nd issue, node failed to boot after the crash:
Looks like timeout for step "Reply commitlog" depends on scylla.yaml parameter **"read_request_timeout_in_ms"**
In my case it was 50s, and node failed to start in +/- the same time. When I increased parameter to 5 minutes, Scylla started as usual.
Commitlog size was ~800Mb, is it ok that Scylla can't handle this size in 50 seconds?

**Comment by slivne** [comment]
@amuzychka 

Thanks for the info

If all clients run a Shema DDL (Create KEYSPACE / Create TABLE) at the same time - it could be that they are racing and that causes the issue (we intend to improve this). Is this what your hadoop clusters do ? 

If so an alternate solution to this - is to create the schema before all the clients connect.

With regards to the second issue we are not aware of commitlog replay issue correlated with read_request_timeout_in_ms. When scylla boots it does have to read from the system tables, and naturally those should not timeout.

We are aware of such an issue of internal system queries timing out under very high load #2462 that was fixed 9c391970b8eeb51bd69ac862a600b78c60b29d06.

To try and replicate the issue it would help if you can share some info on your setup - are you using HDD/SSD/NVME

Can you please provide the contents of /etc/scylla.d/io.conf

**Comment by amuzychka** [comment]
@slivne, thank you.
1st issue:
> If all clients run a Shema DDL (Create KEYSPACE / Create TABLE) at the same time - it could be that they are racing and that causes the issue
> If so an alternate solution to this - is to create the schema before all the clients connect.

Actually, this is what really happens:
There is one orchestrator that run child scripts, each child script run Schema DDL as pre-start step and then start Hadoop cluster (I wrote about 12 as it's max allowed N of clusters for simultaneous work, the rest wait in queue).
The problem was that all child scripts run in subshells simultaneously and pre-start step doesn't have any limitations. As result - DDL queries too (it's about 250 Create KEYSPACE + Create TABLE queries).
So, the real problem **was not related to the Hadoop Cluster**. Scylla was already stopped when Hadoop tried to connect.
Thank you that helped with research.

By the way, Scylla 2.0 works fine under the same load.

2nd issue:

> To try and replicate the issue it would help if you can share some info on your setup

I used public AMI ami-9bc9bd8d and setup it on AWS  i3.2xlarge instance.
Storage type: 1 x 1.9 Tb NVMe SSD
Also, I passed all steps from [here](https://github.com/scylladb/scylla/wiki/AWS-i3)

/etc/scylla.d/io.conf
`SEASTAR_IO="--num-io-queues 6 --max-io-requests 192"`
